### DISCUSSION TITLE
Problem when creating sveltekit project

### DISCUSSION THREAD

**Discussion by crusty-code** [sveltejs/kit#11738] [Q&A]
### Describe the bug

Whenever i try to create a sveltekit project with `pnpm create vite` and use `pnpm install` and then start the dev server with `pnpm run dev` it gives me these errors: `SvelteKitError: Not found: /workbox-c676b6d3.js`, ` SvelteKitError: Not found: /dev-sw.js` and `SvelteKitError: Not found: /@vite-plugin-pwa/pwa-entry-point-loaded`

### Reproduction

use `pnpm create vite` and install the packages with `pnpm install`. Then run the project by using `pnpm run dev`. it gives me the errors 

### Logs

```Shell
"Handling window message"
"Bad source or badly formatted message, skipping."
```

### System Info

```Shell
System:
    OS: Windows 11 10.0.22621
    CPU: (16) x64 AMD Ryzen 7 7730U with Radeon Graphics
    Memory: 3.39 GB / 15.36 GB
  Binaries:
    Node: 20.10.0 - C:\Program Files\nodejs\node.EXE
    pnpm: 8.14.3 - ~\AppData\Local\pnpm\pnpm.EXE
  Browsers:
    Edge: Spartan (44.22621.1702.0)
```

### Severity

blocking all usage of SvelteKit

**Comment by Conduitry** [comment]
Why are you using `pnpm create vite`? That's not a package that's under our control, and there's nothing we can do about how that behaves.

**Comment by teemingc** [comment]
Using `pnpm create vite` and selecting the `svelte` then `SvelteKit` option, I believe it ends up using `create-svelte`. However, I wasn't able to reproduce the issue. Please specify the exact CLI options you have selected.

**Comment by codenoid** [comment]
have you try this command:

```sh
npm create svelte@latest my-app
cd my-app
npm install
npm run dev -- --open
```

**Comment by crusty-code** [comment]
I haven't had a chance to try out if it's a me problem or a problem with sveltekit dropping support for pnpm, but in the past pnpm worked with sveltekit and I loved it. Not only bc it makes it a lot faster to install but also cuz it saves disk space. I feel like many devs would agree with me that not being able to use pnpm is a big deal breaker

**Comment by crusty-code** [comment]
Oops accidentally closed it

**Comment by crusty-code** [comment]
> Using `pnpm create vite` and selecting the `svelte` then `SvelteKit` option, I believe it ends up using `create-svelte`. However, I wasn't able to reproduce the issue. Please specify the exact CLI options you have selected.

I used the vite cli with pnpm then I selected svelte and sveltekit with the demo starter app template with typescript, eslint and prettier. The issue persistedin different directories and when not using eslintand prettier. My pnpm version is 8.14 and I'm using the newest version of create-vite

**Comment by crusty-code** [comment]
> Using `pnpm create vite` and selecting the `svelte` then `SvelteKit` option, I believe it ends up using `create-svelte`. However, I wasn't able to reproduce the issue. Please specify the exact CLI options you have selected.

If you haven't been able to replicate the issue with pnpm than it might be some problem on my device but idk

**Comment by teemingc** [comment]
> I haven't had a chance to try out if it's a me problem or a problem with sveltekit dropping support for pnpm, but in the past pnpm worked with sveltekit and I loved it. Not only bc it makes it a lot faster to install but also cuz it saves disk space. I feel like many devs would agree with me that not being able to use pnpm is a big deal breaker

I believe he meant that we have no control over the `vite` CLI. Is it the same result if you use `pnpm create svelte`? (That is to invoke the create-svelte package directly). You can also try pruning the pnpm cache. I’m not sure why your error logs contain references to vite-plugin-pwa since that’s not included in the sveltekit starter template

**Reply by crusty-code** [reply]
yes i tried `pnpm create svelte` and it threw the same error.

**Reply by crusty-code** [reply]
I have asked other people with pnpm and other people don't report the same issue

**Reply by crusty-code** [reply]
i ran the `pnpm store prune` command after which i wanted to reinstall the dependecies for sveltekit and i got this error ` ERR_PNPM_UNSUPPORTED_ENGINE  Unsupported environment (bad pnpm and/or Node.js version)

Your Node version is incompatible with "/@sveltejs/kit/2.4.3(@sveltejs/vite-plugin-svelte@3.0.1)(svelte@4.2.9)(vite@5.0.12)".

Expected version: >=18.13
Got: v18.5.0

This is happening because the package's manifest has an engines.node field specified.
To fix this issue, install the required Node version.`

**Reply by crusty-code** [reply]
I am currently using node version 20.10.0 with nvm which i have been using until now

**Reply by teemingc** [reply]
Could you run `node -v` and `where node`? Are you running everything in windows (git bash) or WSL?

**Reply by crusty-code** [reply]
node -v gives the before mentioned version v20.10.0. i'm running it in powershell

**Reply by crusty-code** [reply]
I will probably close this discussions since i have just noticed it breakes with other apps too, which indicates its not a sveltekit problem but a problem with vite
